### DISCUSSION TITLE
Wait Until Node - Ignore unavailable/null/NaN?

### DISCUSSION THREAD

**Discussion by pistonhome** [zachowj/node-red-contrib-home-assistant-websocket#598] [Q&A]
Trying to use a wait until node in order to wait until the humidity has dropped in my bathroom. Is there a way to ignore unavailable/null/NaN values (in case the sensor is offline/unavailable). Currently, if unavailable, the node will wait the entire allotted time as it can't fulfill the request and then times out. I tried rigging up some JSONata code, but I am not sure it can be done currently in the wait until node. 

For context, I am trying to wait until the humidity is less than 40%.

![image](https://user-images.githubusercontent.com/38408789/154801643-d1ac2446-4853-4eac-8b9a-864782803cb9.png)

**Comment by zachowj** [comment]
Not sure what you're looking to do when it is one of those values? Trigger right away?

**Comment by pistonhome** [comment]
Great question. In my specific instance, I would want it to trigger TRUE immediately, but in other instances I could see if one preferred it to time out FALSE immediately.

**Reply by zachowj** [reply]
I would recommend using a current-state node before the wait-until node and checking there and then just skipping the wait-until node.

**Reply by pistonhome** [reply]
Yep, i figure that's the only way. Just checking if this happened to be built into wait until nodes. 

Thanks again for your contribution zachow; the wait until node is one of my favorite ones. Especially because it is the only one that can evaluate attributes (as current state node cant).

**Reply by zachowj** [reply]
You can check against attributes in the current state node. It's just not as UI-friendly.

![image](https://user-images.githubusercontent.com/37859597/154820795-f4f48e55-2006-4960-b75b-7d3996c50add.png)

https://zachowj.github.io/node-red-contrib-home-assistant-websocket/guide/jsonata.html

**Reply by pistonhome** [reply]
Ah i didnt even think of that. Good call.
